source/common/ssl/context_impl.h

```diff
@@ -37,8 +36,8 @@ struct SslStats {
   ALL_SSL_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT, GENERATE_TIMER_STRUCT)
 };
 
-typedef CSmartPtr<SSL, SSL_free> SslConPtr;
-typedef CSmartPtr<X509, X509_free> X509Ptr;
+typedef bssl::UniquePtr<SSL> SslConPtr;
+typedef bssl::UniquePtr<X509> X509Ptr;
```

Comment:
Should this still be type aliased? [STYLE.md](https://github.com/lyft/envoy/blob/master/STYLE.md) says so, but I personally find:

    bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl));

more readable than:

    X509Ptr cert = X509Ptr(SSL_get_peer_certificate(ssl));

Also, it seems that we're only type aliasing smart pointers (i.e. `std::unique_ptr<T>`) but not vectors (i.e. `std::vector<T>`), which seems a bit inconsistent.